test/compilable/fix16600.d

```diff
@@ -0,0 +1,17 @@
+// https://issues.dlang.org/show_bug.cgi?id=16600
+
+struct S
+{
+    int i;
+
+    this(string) { i = 1; }
+    this(string) immutable { i = 2; }
+}
+
+void main()
+{
+    auto a = const(S)("abc");
+    assert(a.i == 2);
```

Comment:
Why, why is immutable preferred over mutable here?  They both convert to const, why isn't this ambiguous?